### ISSUE TITLE
[BUG] pygmo interferes with pytorch on Windows system

### ISSUE THREAD

**Issue by CoolRunning** [esa/pygmo2#159]
**Describe the bug**
Installing pytorch and pygmo in the same conda environment makes torch unusable.

**To Reproduce**
In a fresh and empy conda environment on Windows execute in the following order:
```
conda install python=3.9
conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia
```
verify that pytorch works by opening a python console
```
import torch
torch.rand(1)
```
break torch by installing pygmo
```
conda install pygmo
```
verify that pytorch stopped working in a python console
```
import torch
```
**Screenshots**
Output of error in my system:
```
(deepl3) C:\projects>python
Python 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:38:46) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\mmarcus\miniconda3\envs\deepl3\lib\site-packages\torch\__init__.py", line 130, in <module>
    raise err
OSError: [WinError 182] The operating system cannot run %1. Error loading "C:\Users\mmarcus\miniconda3\envs\deepl3\lib\site-packages\torch\lib\fbgemm.dll" or one of its dependencies.
```

**Environment (please complete the following information):**
 - OS: Windows11
 - Installation method: conda
 - Version: 24.3.0

**Additional context**
* bug is reproducible with Python 3.12 as well
* bug is not reproducible in Linux (WSL2, Ubuntu 20.04.6 LTS)
* changing the order to first install pygmo and second install torch does not change the outcome
* A full log of this bug is attached.
[log.txt](https://github.com/esa/pygmo2/files/14847166/log.txt)

**Comment by darioizzo** [comment]
oh no! tricky to investigate .... but https://github.com/pytorch/pytorch/issues/89752 might be related? Maybe more a conda/pytorch issue than a pygmo one?

**Comment by bluescarni** [comment]
Unfortunately I don't think that mixing ``conda-forge`` with other channels is fully supported (though it may work in practice on some setups).

@CoolRunning isn't it possible to install directly torch from ``conda-forge``?

**Comment by CoolRunning** [comment]
I tried installing torch from `conda-forge` with no success, but it might still be possible (did not try very hard). The current workaround that I found is installing pygmo with conda and torch with pip, following [PyTorch](https://pytorch.org/get-started/locally/)

```
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121
```

This allows CUDA as well (in the above case: CUDA 12.1). So far this seems workable, so I close the issue for now.
